server/src/modules/user/user.service.ts

```diff
@@ -33,6 +33,12 @@ export class UserService {
     return mentees;
   }
 
+  async createUser(user: any): Promise<User> {
```

Comment:
nit: see my comment above. In an ideal world we could type this to be `user: CreateUserDTO`, or even consider just taking a list of arguments (like `firstName: string, lastName: string, ...`) and only use DTOs in the controllers.